agent/structs/config_entry_status.go

```diff
@@ -22,7 +22,7 @@ type ResourceReference struct {
 	// unused, this should be blank.
 	SectionName string
 
-	acl.EnterpriseMeta
+	acl.EnterpriseMeta `mapstructure:",squash"`
```

Comment:
Should this also have the `hacl:",squash"` tag that's present on this field for `APIGatewayConfigEntry`, for example?

https://github.com/hashicorp/consul/blob/d5677e5680050a9b22403c160f5edde59613a4fc/agent/structs/config_entry_gateways.go#L713

Reply:
good call, I'll add that